src/Polly/Caching/AsyncCacheEngine.cs

```diff
@@ -3,6 +3,7 @@ namespace Polly.Caching;
 
 internal static class AsyncCacheEngine
 {
+    [DebuggerDisableUserUnhandledExceptions]
```

Comment:
>[DebuggerDisableUserUnhandledExceptions] [](http://example.com/codeflow?start=4&length=40)

Sorry if this is obvious, but just to be clear -- the reason to have this here would be if you expect `cacheProvider` or possibly `onCachePut` to be implemented in user code, and exceptions coming out of user code should be considered ignorable, such as if they will be retried. If that isn't the case, you don't need this even though there is a `catch`

Reply:
All are user-configurable:

https://github.com/App-vNext/Polly/blob/2ff84c8db3b4e6268bd9424a262365c3346c93ed/src/Polly/Caching/AsyncCacheSyntax.cs#L146-L172

Whether they're ignorable would depend on how the user has composed their resilience policy (for example, wrapping a cache policy with retries).